addons/pos_sale/tests/test_pos_sale_flow.py

```diff
@@ -622,3 +623,42 @@ def test_downpayment_amount_to_invoice(self):
         # Confirm all invoices
         sale_order.invoice_ids.action_post()
         self.assertEqual(sale_order.order_line[1].price_unit, 10)
+
+    def test_downpayment_line_name(self):
+        product_a = self.env['product.product'].create({
+            'name': 'Product A',
+            'available_in_pos': True,
+            'type': 'product',
+            'lst_price': 100.0,
+            'taxes_id': [],
+        })
+        partner_test = self.env['res.partner'].create({'name': 'Test Partner'})
+
+        sale_order = self.env['sale.order'].create({
+            'partner_id': partner_test.id,
+            'order_line': [(0, 0, {
+                'product_id': product_a.id,
+                'name': product_a.name,
+                'product_uom_qty': 1,
+                'product_uom': product_a.uom_id.id,
+                'price_unit': product_a.lst_price,
+            })],
+        })
+        sale_order.action_confirm()
+
+        self.downpayment_product = self.env['product.product'].create({
+            'name': 'Down Payment',
+            'available_in_pos': True,
+            'type': 'service',
+            'taxes_id': [],
+        })
+        self.main_pos_config.write({
+            'down_payment_product_id': self.downpayment_product.id,
+        })
+        self.main_pos_config.open_ui()
+        self.start_tour("/pos/ui?config_id=%d" % self.main_pos_config.id, 'PoSDownPaymentAmount', login="accountman")
+
+        downpayment_line_pos = sale_order.order_line.filtered(lambda line: self.downpayment_product.name in line.name)
```

Comment:
```suggestion
        downpayment_line_pos = sale_order.order_line.filtered('is_downpayment')
```
Name comparison is not the most reliable way most of the time, so avoid it when possible ;).
(you could also have compared the product to the down payment product).